### PR TITLE
[feat] Fan club join/leave/request logic implemented

### PR DESCRIPTION
**Pull request by rado196**
This PR includes:

Already implemented:
- Join to fan club (if the fan club is public).
- Request to join (if the fan club is private).
- Leave the fan club.
- Accept invitation.
- Reject invitation.

### CODE REVIEW

File: clients/app/src/themes/theming/components/Text.ts [hunk 1]
```diff
@@ -291,5 +291,42 @@ export default {
         },
       };
     },
+    actionSheetTitle: ({ colorMode }: ComponentParams) => {
```
Comment (hay92k) [inline clients/app/src/themes/theming/components/Text.ts:294]: we don't need new text variants

File: clients/app/src/views/components/compositions/FanClubInfo/FanClubJoinButton.tsx [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+
+  /**
+   * On join button press.
+   *
+   * If fan club is public then user will be joined to fan club automatically,
+   * otherwise join request will be sent and state will be in pending-invitation.
+   */
+  const onJoinPress = React.useCallback(() => {
+    const newJoinStatus =
+      data.privacy === FanClubPrivacyEnum.public
```
Comment (hay92k) [inline clients/app/src/views/components/compositions/FanClubInfo/FanClubJoinButton.tsx:47]: Isn't it better to use if .. else ?

File: clients/app/src/views/components/compositions/FanClubInfo/FanClubJoinButton.tsx [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+
+    fanClubMembersStore.setAddFieldValue('fanClubId', data.id);
+    fanClubMembersStore.create();
+  }, [data.privacy, data.id, fanClubMembersStore]);
+
+  /**
+   * On leave button press.
+   */
+  const onLeavePress = React.useCallback(() => {
+    setJoinStatus(undefined);
```
Comment (hay92k) [inline clients/app/src/views/components/compositions/FanClubInfo/FanClubJoinButton.tsx:61]: better to have status, not undefined

File: clients/app/src/views/components/compositions/FanClubInfo/actions/CancelRequestActionSheet.tsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
+import React from 'react';
```
Comment (hay92k) [inline clients/app/src/views/components/compositions/FanClubInfo/actions/CancelRequestActionSheet.tsx:1]: Can we combine this [CancelRequestActionSheet.tsx] and [RespondInvitationActionSheet.tsx] ?

File: clients/app/src/views/components/compositions/FanClubInfo/FanClubJoinButton.tsx [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+    onRespondInvitationPress,
+    joinStatus,
+    theme.colors,
+  ]);
+
+  return (
+    <>
+      <Button
+        leftIcon={
+          !icon ? undefined : <Icon name={icon.name} color={icon.color} size="ic-xs" />
```
Comment (hay92k) [inline clients/app/src/views/components/compositions/FanClubInfo/FanClubJoinButton.tsx:178]: isn't `icon && <Icon name={icon.name} color={icon.color} size="ic-xs" />` better?

**Review by hay92k — CHANGES_REQUESTED** [review]

File: clients/app/src/themes/theming/components/Text.ts (hunk 1 above)
Comment (rado196) [reply clients/app/src/themes/theming/components/Text.ts:294]: ok, will revert

File: clients/app/src/views/components/compositions/FanClubInfo/FanClubJoinButton.tsx (hunk 2 above)
Comment (rado196) [reply clients/app/src/views/components/compositions/FanClubInfo/FanClubJoinButton.tsx:47]: will replace to if-else :)

File: clients/app/src/views/components/compositions/FanClubInfo/FanClubJoinButton.tsx (hunk 3 above)
Comment (rado196) [reply clients/app/src/views/components/compositions/FanClubInfo/FanClubJoinButton.tsx:61]: updated

File: clients/app/src/views/components/compositions/FanClubInfo/FanClubJoinButton.tsx (hunk 5 above)
Comment (rado196) [reply clients/app/src/views/components/compositions/FanClubInfo/FanClubJoinButton.tsx:178]: yeah, it's better :)

File: clients/app/src/views/components/compositions/FanClubInfo/actions/CancelRequestActionSheet.tsx (hunk 4 above)
Comment (rado196) [reply clients/app/src/views/components/compositions/FanClubInfo/actions/CancelRequestActionSheet.tsx:1]: yeah, updated.

File: apis/core-api/src/core/data/models/FanClubMember.ts [hunk 6]
```diff
@@ -114,6 +114,7 @@ module.exports = (sequelize: Sequelize): typeof FanClubMember => {
       status: {
         type: DataTypes.ENUM({
           values: [
+            FanClubMemberStatusEnum.noStatus,
```
Comment (hay92k) [inline apis/core-api/src/core/data/models/FanClubMember.ts:117]: maybe notRelated?

File: clients/app/src/views/components/compositions/FanClubInfo/FanClubJoinButton.tsx [hunk 7]
```diff
@@ -125,43 +123,52 @@ const FanClubJoinButton: React.FC<IFanClubInfoProps> = ({ data }) => {
...
-        return buildButtonAttributes(joinStatus, theme.colors, onLeavePress);
-      case FanClubMemberStatusEnum.pendingRequest:
-        return buildButtonAttributes(joinStatus, theme.colors, onCancelRequestPress);
-
-      case FanClubMemberStatusEnum.pendingInvitation:
-        return buildButtonAttributes(joinStatus, theme.colors, onRespondInvitationPress);
-
-      case FanClubMemberStatusEnum.banned:
-        return buildButtonAttributes(joinStatus, theme.colors);
+    if (joinStatus === FanClubMemberStatusEnum.pendingRequest) {
```
Comment (hay92k) [inline clients/app/src/views/components/compositions/FanClubInfo/FanClubJoinButton.tsx:162]: don't we need else if-s here?

**Review by hay92k — CHANGES_REQUESTED** [review]

File: clients/app/src/views/components/compositions/FanClubInfo/FanClubJoinButton.tsx (hunk 7 above)
Comment (rado196) [reply clients/app/src/views/components/compositions/FanClubInfo/FanClubJoinButton.tsx:162]: each `if` condition returns value, so using else is unnecessary.

File: apis/core-api/src/core/data/models/FanClubMember.ts (hunk 6 above)
Comment (rado196) [reply apis/core-api/src/core/data/models/FanClubMember.ts:117]: okay, will change now.

**Review by hay92k — APPROVED** [review]
lgtm

**Pull request merged**